### ISSUE TITLE
Stable IDs across placetype-changes?

### ISSUE THREAD

**Issue by frankieroberto** [whosonfirst-data/whosonfirst-data#46]
Hello. I'm posting this here as more of a question/challenge/discussion point than an issue/bug – if there's a better place for this kind of feedback let me know.

Basically, I'd like to poke at this a bit:

> A location’s place type is a pretty fundamental property and it’s easy to imagine that a third-party application might use and store that metadata for important application-specific purposes. Which is to say: We don’t need to have an opinion about what or why an application is doing something with the metadata associated with a location and if we start by saying that WOF ID 85892915 is a neighbourhood (which it was when we imported the data from Quattroshapes) we probably shouldn’t just change it on a whim.

It's probable that you’ve thought more about this than I have, but I can see an argument for being able to keep the same ID whilst changing a place's position in the place-type hierarchy IF it's still conceptually the same place. The two that come to mind are: a) a place succeeding into a place-type higher up in the hierarchy (e.g. a region becoming a country) and b) correcting 'mistakes' in the place classifications (where mistake might mean changing your mind about how your fixed hierarchy applies to the messy real way that places get organised).

Real world example: [Rutland](https://en.wikipedia.org/wiki/Rutland#History) in the UK was a county, but got demoted to a district of Leicestershire in 1974 but then promoted again to a country (technically a 'unitary authority') in 1997. Throughout this though it retained the same shape, local identity, etc, and thus was arguably still conceptually the *same* place.

Which basically means that I'm not sure I agree that a place's place-type is 'fundamental' (perhaps it's more 'incidental'?). This is possibly the influence of being from the UK where the organisation of places is byzantine, little-understood, and subject to change every decade or so. (Our understanding of 'country' is different from everyone else's, and that's just for starters: the local authority structure differs depending on which part of the UK you're talking about and whether you're interested in responsibility for bin collection, healthcare or policing).

Of course, what this really boils down to is what kind of authority/opinion Mapzen wants to make about places. Is the gazetteer a record of 'official' places and how they are/have been organised politically, or is it a record of the social construct of places? I'd be more interested in the latter but I can see use-cases for both.

A final point is to consider what happens when a place retains the same place-type but changes its parent: would this automatically necessitate a new succeeding ID in the same way? Real world example: Monmouthshire in the UK has flipped back and forth between being part of England and Wales a couple of times (and was even ambiguous for a while). I'd argue not, and that the same should apply to placetype-changes.

**Comment by nvkelso** [comment]
Hi @frankieroberto. Thanks for your thoughtful question. We've started to employ **wof_id** changes via **supersedes** and **superseded-by** pointers when a feature changes place type (would go into a different meta file) or changes hierarchy (and all it's children would need to update hierarchy). 

So far this is primarily related to cleaning up the data and also a desire to exercise our internal code and downstream code to allow for these types of ID daisy chains. 

/cc @stepps00

**Comment by stepps00** [comment]
Hey @frankieroberto - I'm following up on your initial comments about stable wof:ids in regards to changing wof:placetypes..

First off, we've recently created `wof:id` documentation, which can be found [here](https://github.com/whosonfirst/whosonfirst-cookbook/blob/master/definition/wof:id_lifecycle.md). This life cycle documentation was created to address comments and concerns similar to yours - give it a read when you have a few moments. Your example about Rutland is well taken, but let me expand on that example given our `wof:id` life cycle rules...

Though Rutland is the "same place," Who's On First _should_ be viewed as taking a snapshot in time to capture: Rutland as a county (pre-1974), Rutland as a district (1974-1997), and Rutland as a county (post-1997). If the `wof:id` was kept the same throughout all of these changes, Who's On First wouldn't be able to display Rutland as a "place" during each of these three, distinct time periods. By creating new records (each linked through `wof:superseded` and `wof:superseded_by` fields), the history of that place is preserved. Even though Rutland kept it's same size, identity, etc., it was still a different "place" in, say, 1984 as compared to 2003.

To your second point - changing a `wof:parent` id would also warrant a new record with superseding taking place. This will cause churn in Who's On First, but as long as our superseding work is done, the history of these places can be found when viewing the current record of that place.

Thanks for your patience on this issue while we hammered out the life cycle rules - take a look at the documentation and let us know your thoughts!
